code/modules/projectiles/updated_projectiles/guns/rifles.dm

```diff
@@ -406,3 +406,45 @@
 	recoil_unwielded = CONFIG_GET(number/combat_define/high_recoil_value)
 
 //-------------------------------------------------------
+//MK224
+
+/obj/item/weapon/gun/rifle/autoshotgun
+	name = "\improper MK224 Tactical Shotgun"
+	desc = "A modified version of the MK221 Tactical Shotgun, now including a magazine."
+	icon_state = "mk224"
+	item_state = "mk221"
+	fire_sound = "sound/weapons/gun_shotgun_automatic.ogg"
+	current_mag = /obj/item/ammo_magazine/rifle/autoshotgun
+	attachable_allowed = list(
+						/obj/item/attachable/quickfire,
+						/obj/item/attachable/bayonet,
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/verticalgrip,
+						/obj/item/attachable/angledgrip,
+						/obj/item/attachable/lasersight,
+						/obj/item/attachable/gyro,
+						/obj/item/attachable/flashlight,
+						/obj/item/attachable/extended_barrel,
+						/obj/item/attachable/heavy_barrel,
+						/obj/item/attachable/magnetic_harness,
+						/obj/item/attachable/stock/tactical,
+						/obj/item/attachable/attached_gun/grenade,
+						/obj/item/attachable/attached_gun/flamer,
+						/obj/item/attachable/scope/mini)
```

Comment:
Check each attachment, and see if the sprite looks good with them. It's unlikely to be true for several of them.
Also consider which one makes sense. Does a gyro make sense here? Do you want it to be shot one-handed? What about a laser sight or a quick fire?